# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/antitracking/ContentBlocking.h
# Commit: 3b1dbe909aaf
# Full Hash: 3b1dbe909aaf1982c72db1e5bc8cb35208ae6c77
# Author: Dimi Lee <dlee@mozilla.com>
# Date: 2020-04-23 21:43:09
# Regressor Bug: 1616775
# File Overlap Count: 1
# Description:
#   Bug 1616775 - P4. Add CompleteAllowAccessFor IPDL r=timhuang,baku
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D71013
# ==============================================================================

diff -r 123b6c3a0ede -r 3b1dbe909aaf toolkit/components/antitracking/ContentBlocking.h
--- a/toolkit/components/antitracking/ContentBlocking.h	Thu Apr 23 14:28:41 2020 +0000
+++ b/toolkit/components/antitracking/ContentBlocking.h	Thu Apr 23 14:25:16 2020 +0000
@@ -27,6 +27,7 @@
 
 namespace dom {
 class BrowsingContext;
+class ContentParent;
 }
 
 class ContentBlocking final {
@@ -109,6 +110,7 @@
           StaticPrefs::privacy_restrict3rdpartystorage_expiration());
 
  private:
+  friend class dom::ContentParent;
   // This should be running either in the parent process or in the child
   // processes with an in-process browsing context.
   static MOZ_MUST_USE RefPtr<StorageAccessGrantPromise> CompleteAllowAccessFor(
